apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+      // Act
+      const result = caller.readPageAndBlob({
+        siteId: 1,
+        pageId: 1,
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "NOT_FOUND", message: "Resource not found" }),
+      )
+    })
+
+    it("should return the resource if resource type is Page and exists", async () => {
+      // Arrange
+      const { site, page, blob, navbar, footer } = await setupPageResource({
+        resourceType: "Page",
+      })
+      const expected = {
+        ...pick(page, ["permalink", "title", "type"]),
+        navbar: omit(navbar, ["createdAt", "updatedAt"]),
+        footer: omit(footer, ["createdAt", "updatedAt"]),
+        content: blob.content,
+      }
+
+      // Act
+      const result = await caller.readPageAndBlob({
+        siteId: site.id,
+        pageId: Number(page.id),
+      })
+
+      // Assert
+      expect(result.type).toEqual("Page")
+      expect(result).toMatchObject(expected)
+    })
+
+    it("should return the resource if resource type is RootPage and exists", async () => {
+      // Arrange
+      const { site, page, blob, navbar, footer } = await setupPageResource({
+        resourceType: "RootPage",
+      })
+      const expected = {
+        ...pick(page, ["permalink", "title", "type"]),
+        navbar: omit(navbar, ["createdAt", "updatedAt"]),
+        footer: omit(footer, ["createdAt", "updatedAt"]),
+        content: blob.content,
+      }
+
+      // Act
+      const result = await caller.readPageAndBlob({
+        siteId: site.id,
+        pageId: Number(page.id),
+      })
+
+      // Assert
+      expect(result.type).toEqual("RootPage")
+      expect(result).toMatchObject(expected)
+    })
+
+    it("should return the resource if resource type is CollectionPage and exists", async () => {
+      // Arrange
+      const { site, page, blob, navbar, footer } = await setupPageResource({
+        resourceType: "CollectionPage",
+      })
+      const expected = {
+        ...pick(page, ["permalink", "title", "type"]),
+        navbar: omit(navbar, ["createdAt", "updatedAt"]),
+        footer: omit(footer, ["createdAt", "updatedAt"]),
+        content: blob.content,
+      }
+
+      // Act
+      const result = await caller.readPageAndBlob({
+        siteId: site.id,
+        pageId: Number(page.id),
+      })
+
+      // Assert
+      expect(result.type).toEqual("CollectionPage")
+      expect(result).toMatchObject(expected)
+    })
+
+    it("should return 404 if resource type is not a page", async () => {
+      // Arrange
+      const { site, folder } = await setupFolder()
+
+      // Act
+      const result = caller.readPageAndBlob({
+        siteId: site.id,
+        pageId: Number(folder.id),
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "NOT_FOUND", message: "Resource not found" }),
+      )
+    })
+  })
+
+  describe("reorderBlock", () => {
```

Comment:
sorry for burden  - this test case isn't required for this pr but if you decide to not do it, could you file an issue and assign it to me?

we should check the behaviour in the event of value equality but different ordering of properties. as an example, consider the state `[{a:1, b:2}, some_other_block]`. if this gets changed to `[{b:2, a:1}, some_other_block]`, should we consider the state as being changed? to the user, this will appear exactly the same (same properties on block and block is same index) but i think this will fail atm cos we do a string comparison